bitnami/contour/templates/contour/deployment.yaml

```diff
@@ -110,8 +110,13 @@ spec:
             {{- if .Values.contour.ingressStatusAddress }}
             - --ingress-status-address={{ .Values.contour.ingressStatusAddress }}
             {{- else }}
+            {{- if .Values.envoy.service.multiAz.enabled }}
             - --envoy-service-namespace={{ default .Release.Namespace .Values.contour.envoyServiceNamespace }}
-            - --envoy-service-name={{ default (printf "%s-envoy" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-") .Values.contour.envoyServiceName }}
+            - --envoy-service-name={{ default (printf "%s-%s-envoy" (include "common.names.fullname" .) (index .Values.envoy.service.multiAz.zones 0).name | trunc 63 | trimSuffix "-") .Values.envoy.service.name }}
+            {{- else }}
+            - --envoy-service-namespace={{ default .Release.Namespace .Values.contour.envoyServiceNamespace }}
+            - --envoy-service-name={{ default (printf "%s-envoy" (include "common.names.fullname" .) | trunc 63 | trimSuffix "-") .Values.envoy.service.name }}
```

Comment:
Why not using a ternary operator to define this flag? 

If not, just configure the if/else block with this flag (remove the --envoy-service-namespace one as I mentioned above)

Reply:
I just noticed that we have .Values.envoy.service.name and .Values.contour.envoyServiceName. You are moving to the former option so I think we need to deprecate the latter. To do so, please add a DEPRECATED note in the values.yaml file (there are some examples in the values.yaml already) and update the deployment to use this code (or a similar one)

```
            {{- $envoyServiceName := coalesce .Values.envoy.service.name .Values.contour.envoyServiceName }}
            {{- if .Values.envoy.service.multiAz.enabled }}
            - --envoy-service-name={{ default (printf "%s-%s-envoy" (include "common.names.fullname" .) (index .Values.envoy.service.multiAz.zones 0).name | trunc 63 | trimSuffix "-")  $envoyServiceName }}
            {{- else }}
            ...
```

Reply:
No worries, thanks for the tip and the line of code I can use. Otherwise, I probably had to try and error some hours.